pkg/build/droneutil/event.go

```diff
@@ -0,0 +1,34 @@
+package droneutil
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+// Lookup is the equivalent of os.LookupEnv,  but also accepts a list of strings rather than only checking os.Environ()
+func Lookup(values []string, val string) (string, bool) {
```

Comment:
I am not sure about this change, I'd like to see a 1:1 match between what you want and what you expect rather than giving a list of env vars and doing checks tbh. But not a strong preference there!

Reply:
Yeah it's a little weird. The way you'd use this is just by calling it with `Lookup(os.Environ(), "DRONE_BUILD_NUMBER")`.

Doing it this way means we don't have to set and unset environment variables in tests. Those kinds of operations make for flakey tests.